### PR TITLE
Added CSH Bootstrap

### PR DESCRIPTION
**Pull request by akaStanley**
image
image
image

### CODE REVIEW

File: routes.go
Comment (stevenmirabito) [inline routes.go]: Why is this still embedded in a Go executable? 😬

Comment (liam-middlebrook) [reply routes.go]: Agreed. I’ll take some time later tonight to write up a more complete review, but I think that this change should include the full switch to templates as illustrated in `view_plugs.tmpl`.

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl]: On routes.go:100 you pass in the plug's S3ID as `plug_s3id`, but you do not pass in a `plug` object.

File: routes.go
Comment (liam-middlebrook) [inline routes.go]: In success.tmpl:57 you try to access a `plug` object in order to get a S3 pre-signed URL. It looks like that's the only dynamic data you're grabbing from that page, so I think it would make more sense to just pass the pre-signed URL directly into the template at this point.

File: templates/upload.tmpl
Comment (liam-middlebrook) [inline templates/upload.tmpl]: In routes.go:60 we're currently grabbing the input by it's `name` attribute (I thought it was `id`, but looking through [this](https://github.com/gin-gonic/gin/blob/master/examples/upload-file/single/public/index.html) Gin example makes me think otherwise. I'd have to check the source to be 100% certain though), we're looking for a value of `file`, which you currently have not defined. Since `file` is ambiguous maybe it would be a good idea to also edit that line so that we're looking for something a bit more clear, like `fileUpload` or `uploadedPlug`.

File: templates/upload.tmpl
Comment (liam-middlebrook) [inline templates/upload.tmpl]: I'd prefer if this was a link to the ComputerScienceHouse/CodeOfConduct repository rather than a link to a PDF on a third-party site.

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl]: Please ensure your files end with newline characters, here's a StackOverflow post that links to the POSIX standard and provides a pretty solid explanation behind why this is important: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl:2]: While I'd like to see this content not repeated across different templates, it looks like Gin doesn't currently support that out of the box: https://github.com/gin-gonic/gin#html-rendering

If that's something you're willing to take on as part of this PR that would be awesome, but it's not something that I'll block merging it because of.

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl]: I think it would be nice if you could move this to a static CSS file. Maybe make a directory in the same level as `templates` called `static`?

Here's an example of how to serve static files with Gin: https://github.com/gin-gonic/gin#serving-static-files

Be careful with the paths, when I was writing out the Openshift container image for Golang+Gin I had to add some special environment variable to get the paths right. You can find that usage on main.go:30

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl]: Nitpick: Please keep your comments above the line they relate to if they don't fit to it's side (and stay under 80 characters in length).

**Review by liam-middlebrook — CHANGES_REQUESTED** [review]
Looks good so far, but there's still a bit of work to go before it's ready to merge.

Thanks for checking the SHA-384's that was a nice touch 🥇

File: templates/success.tmpl
Comment (liam-middlebrook) [inline templates/success.tmpl]: I forgot to ask, is this a copy-paste artifact?

File: templates/success.tmpl
Comment (sgreene570) [reply templates/success.tmpl:2]: +1 Templating would be really cool, but not the end of the world if we don't have it I suppose.  If it doesn't get added in this PR it should be submitted as a project issue once this is merged.

File: templates/success.tmpl
Comment (sgreene570) [reply templates/success.tmpl]: In line styles aren't a fun time, especially when its more than one quick thing.

File: templates/success.tmpl
Comment (sgreene570) [inline templates/success.tmpl]: Might be a good idea to give the image tag an id or class and set the css separately.

File: templates/success.tmpl
Comment (sgreene570) [reply templates/success.tmpl]: Also yes please for all things good use new lines at the end of files.  If it stays as is, the next person to modify this file with a sane editor will have to commit the correction.

File: templates/upload.tmpl
Comment (sgreene570) [inline templates/upload.tmpl]: Copy pasta not needed if you set a CSS file.

File: templates/upload.tmpl
Comment (sgreene570) [inline templates/upload.tmpl:72]: Nice job putting script calls at the bottom of the page 👌

File: templates/view_plugs.tmpl
Comment (sgreene570) [inline templates/view_plugs.tmpl:26]: Keep blank line spacing consistent when possible.

**Review by sgreene570 — CHANGES_REQUESTED** [review]
Almost there, a few more things would be awesome though.

**Review by liam-middlebrook — APPROVED** [review]

**Pull request merged**
